cli/main.c

```diff
@@ -157,33 +145,62 @@ int main(int argc, char** argv) {
   // FIXME: this is temp till arg parse implemented.
   PkResult result;
 
-  if (argc == 1) {
+  static const char* const usage[] = {
+    "pocket [option] ... [-c cmd | file | -] [arg] ...",
+    NULL,
+  };
+
+  bool debug = false, quiet = false, version = false, help = false;
+  const char** cmd = NULL;
+
+  struct argparse_option cli_opts[] = {
+    OPT_STRING('c', "cmd", &cmd, "Evaluate and run the passed string."),
+    OPT_BOOLEAN('d', "debug", &debug, "Compile and run a debug version."),
+    OPT_BOOLEAN('h', "help", &help, "Prints this help message and exit."),
+    OPT_BOOLEAN('q', "quiet", &quiet,
+               "Don't print version and copyright statement on REPL startup."),
+    OPT_BOOLEAN('v', "version", &version,
+               "Prints the pocketlang version number and exit."),
+    OPT_END(),
+  };
+
+  struct argparse argparse;
+  argparse_init(&argparse, cli_opts, usage, 0);
+  argc = argparse_parse(&argparse, argc, argv);
+
+  if (debug) {
+    options.debug = debug;
     options.repl_mode = true;
     repl(vm, &options);
-
-  } if (argc >= 3 && strcmp(argv[1], "-c") == 0) {
-
+  } else if (help) argparse_usage(&argparse);
+  else if (version) printf("pocketlang %s\n", PK_VERSION_STRING);
+  else if (cmd) {
     PkStringPtr source = { argv[2], NULL, NULL };
     PkStringPtr path = { "$(Source)", NULL, NULL };
-
     result = pkInterpretSource(vm, source, path, NULL);
     pkFreeVM(vm);
     return result;
-  }
-
-  PkStringPtr resolved = resolvePath(vm, ".", argv[1]);
-  PkStringPtr source = loadScript(vm, resolved.string);
+  } else if (argc >= 1 && argv[0] != NULL) {
+    PkStringPtr resolved = resolvePath(vm, ".", argv[0]);
+    PkStringPtr source = loadScript(vm, resolved.string);
+
+    if (source.string != NULL) {
+      result = pkInterpretSource(vm, source, resolved, &options);
+    } else {
+      result = PK_RESULT_COMPILE_ERROR;
+      fprintf(stderr, "Error: cannot open file at \"%s\"\n", resolved.string);
+      if (resolved.on_done != NULL) resolved.on_done(vm, resolved);
+      if (source.on_done != NULL) source.on_done(vm, source);
+    }
 
-  if (source.string != NULL) {
-    result = pkInterpretSource(vm, source, resolved, &options);
-
-  } else {
-    result = PK_RESULT_COMPILE_ERROR;
-    fprintf(stderr, "Error: cannot open file at \"%s\"\n", resolved.string);
-    if (resolved.on_done != NULL) resolved.on_done(vm, resolved);
-    if (source.on_done != NULL) source.on_done(vm, source);
+    return result;
+  } else if (quiet) TODO;
+  else {
+    options.repl_mode = true;
```

Comment:
Currently we're calling `repl(vm, &options)` at 2 places. Here and `debug` is true. We should only call it one place, and it's when the `argc == 0`

Running a file with the following command below would start the REPL then actually running it.
```
./pocket file.pk a b c
```